add support for equal_nan in torch.testing.assert_close

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56373 add `torch.testing` to docs
* #56365 Split `test_testing.py::TestAsserts` for multiple devices
* #55890 enable support for custom error messages in `torch.testing`
* **#55788 add support for equal_nan in torch.testing.assert_close**
* #55786 Add support for scalars and numpy in torch.testing
* #55385 Add support for checking tensor containers in `torch.testing`

Differential Revision: [D27903821](https://our.internmc.facebook.com/intern/diff/D27903821)

pmeier committed:
add support for equal_nan in torch.testing.assert_close

[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

mruberry reviewed (approved):
Cool! We'll need to think more about the complex case when we add support for it.

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"




[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"


Differential Revision: [D27903821](https://our.internmc.facebook.com/intern/diff/D27903821)

[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"


Differential Revision: [D27903821](https://our.internmc.facebook.com/intern/diff/D27903821)

[ghstack-poisoned]

pmeier committed:
Update on "add support for equal_nan in torch.testing.assert_close"


Differential Revision: [D27903821](https://our.internmc.facebook.com/intern/diff/D27903821)

[ghstack-poisoned]
